Gm streaming

GabrieleManduchi opened the pull request:
Added support for data streaming. node.js event_server ptogram will receive data events and send associated samples to Web pages via ServerSentEvent.

GabrieleManduchi committed:
Feature:First implementation of Server Sent Event implementation of data streaming visualization

Class EventStream has been added to C++ MdsObject for sending one or more samples embedded in a MDSplus message. nodejs/event_server is  a node.js application for serving Web pages for visualization and dispatching streaming events to the registered listener clients.

GabrieleManduchi committed:
Feature:First implementation of Server Sent Event implementation of data streaming visualization

Class EventStream has been added to C++ MdsObject for sending one or more samples embedded in a MDSplus message. nodejs/event_server is  a node.js application for serving Web pages for visualization and dispatching streaming events to the registered listener clients.

GabrieleManduchi committed:
Added streamning server

GabrieleManduchi committed:
Added streamning server

GabrieleManduchi committed:
Feature: Add streaming Device for MARTe2

A MARTE2_OUT device has been added to wrap MDSplus streaming into the MDSplus-MARTe2 device infrastructure

GabrieleManduchi committed:
Feature: Add streaming Device for MARTe2

A MARTE2_OUT device has been added to wrap MDSplus streaming into the MDSplus-MARTe2 device infrastructure

GabrieleManduchi committed:
Feature:Support for data streaming

Added required components for Web based data streaming

GabrieleManduchi committed:
Feature:Support for data streaming

Added required components for Web based data streaming

GabrieleManduchi committed:
Feature:pass listen port for events and SSE to node.js event_server

GabrieleManduchi committed:
Feature:pass listen port for events and SSE to node.js event_server

AndreaRigoni committed:
Fix:fix compile errors in for datastreams.cpp that accurr in updatepkg

AndreaRigoni committed:
Fix:remove old versions

AndreaRigoni committed:
Fix:remove old versions

zack-vii commented:
@GabrieleManduchi : Could you attend the following two things.
1. There seems to be a problem with cpp code under Windows (missing entry in .def perhaps):
```log
17:05:11 mdstreeobjects.obj : error LNK2001: unresolved external symbol "public: virtual void __cdecl MDSplus::EventStream::run(void) __ptr64" (?run@EventStream@MDSplus@@UEAAXXZ)
17:05:11 \workspace\releasebld\64\bin_x86_64\MdsObjectsCppShr-VS.dll : fatal error LNK1120: 1 unresolved externals
```
2. Could you please rebase your branch.
```git
git fetch upstream/alpha
git rebase  upstream/alpha -i
# check if all good
git push -f
```

GabrieleManduchi commented:
retest this please

AndreaRigoni committed:
Fix:remove unresolved symbols

Include atastreams ion VS project in order to remove the unresolved symbols

zack-vii commented:
@AndreaRigoni : problem persists

GabrieleManduchi commented:
retest this please

GabrieleManduchi commented:
retest this please

GabrieleManduchi commented:
From console all tests passed, the procedure appears have timed out

AndreaRigoni committed:
Fix: Avoid undefined ref in Visual Studio build

Since the datastreams.cpp code is incompatible with Visual Studio
the EventStream class declaration must be enclosed in an #ifndef WINDOWS_H

This commit also removes several obsolete scripts and Visual Studio
project files which were only used when doing Visual Studio builds
on a windows server.

tfredian commented:
Gabriele the build is failing as Timo mentioned from an undefined reference to the run() method.

00:11:49.228 mdsdataobjects.obj : error LNK2001: unresolved external symbol "public: virtual void __cdecl MDSplus::EventStream::run(void) __ptr64" (?run@EventStream@MDSplus@@UEAAXXZ)

I'm far from knowledgeable on c++ but shouldn't the  run method in class EXPORT EventStream:public Event in mdsobjects.h be declared as:

    virtual void run() {}

like it is in class EXPORT Event?

This undefined reference error is happening during the VS build of the dll.

The error can be found at: http://jenkins2.mdsplus.org/job/windows/3142/console

GabrieleManduchi commented:
Just removed from Datastreams class

GabrieleManduchi commented:
retest this please

GabrieleManduchi commented:
......not sure the last test referred to the last push

zack-vii reviewed (changes requested):
I think you should put the EXPORT of EventStream::run back for non WINDOWS_H otherwise you might not be able to use the stream. the last commit was the real fix i think

zack-vii commented on mdsobjects/cpp/datastreams.cpp:
```diff
@@ -431,7 +431,7 @@ EXPORT void EventStream::send(int shot, const char *name, int numSamples, uint64
 
 
 
-EXPORT void EventStream::run()
```
I think you should put the export back for non WINDOWS_H otherwise you might not be able to use the stream

zack-vii commented:
can i do one last cleanup and then you may merge it.
